### PR TITLE
Depends on `tzdata` on Windows

### PR DESCRIPTION
**Pull request by jjerphan**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

File: recipe/bld.bat [hunk 1]
```diff
@@ -1,4 +1,13 @@
-cmake -G "NMake Makefiles" -D CMAKE_INSTALL_PREFIX=%LIBRARY_PREFIX% %SRC_DIR% -D BUILD_SHARED_LIBS=ON -D BUILD_TZ_LIB=ON -D USE_SYSTEM_TZ_DB=ON -D CMAKE_BUILD_TYPE=Release
+export CXXFLAGS="$CXXFLAGS -DINSTALL=$PREFIX\share\zoneinfo\"
+
+cmake -G "NMake Makefiles" ^
+     -D CMAKE_INSTALL_PREFIX=%LIBRARY_PREFIX% ^
+     %SRC_DIR% ^
+     -D BUILD_SHARED_LIBS=ON ^
+     -D BUILD_TZ_LIB=ON ^
+     -D USE_SYSTEM_TZ_DB=ON ^
```
Comment (jjerphan) [inline recipe/bld.bat:9]: Unrelated: `USE_SYSTEM_TZ_DB` is unsupported on Windows, it seems: https://github.com/HowardHinnant/date/pull/611

cc @pidefrem

**Comment by jjerphan** [comment]
@conda-forge-admin, please rerender

**Comment by jjerphan** [comment]
@conda-forge-admin, please rerender

**Comment by jjerphan** [comment]
We need another approach, because prefix replacement has been removed in Windows.

[dlls on Windows can access their path](https://stackoverflow.com/a/6924332) (pointer courtesy of @SylvainCorlay); from there we can patch [`access_install`](https://github.com/HowardHinnant/date/blob/1ead6715dec030d340a316c927c877a3c4e5a00c/src/tz.cpp#L398-L419).

**Comment by jjerphan** [comment]
This depends on https://github.com/conda-forge/staged-recipes/pull/26481.

**Comment by SylvainCorlay** [comment]
Closing as superseeded by next PR.

**Pull request closed without merging**